pkg/pmem-csi-driver/nodeserver.go

```diff
@@ -42,7 +43,12 @@ type nodeServer struct {
 	nodeCaps []*csi.NodeServiceCapability
 	cs       *nodeControllerServer
 	// Driver deployed to provision only ephemeral volumes(only for Kubernetes v1.15)
-	mounter mount.Interface
+	mounter       mount.Interface
+	volReadOnly   map[string]bool
```

Comment:
How about introducing a new structure and keeping all these new params as fields in it. Then we can maintain a single `VolumeId:Params` map, easy to maintain. If I remember correctly we used to have such type earlier but removed as found no real need that time.

Reply:
yes, good idea. I started with that, but I realized now the creation and deletion happen in two places, so we can not just conveniently get rid of whole map entry in one place.
Another related question arises: I added map for fsType, which is there now for completeness, but is not actually used. This part is created and used in different function.
Can we perhaps leave fsType out, the we can have one map handling in single place.

Reply:
> creation and deletion happen in two places, so we can not just conveniently get rid of whole map entry in one place.

Yes, If I understood correctly the change, the two possible places depend on type of volume: ephemeral or persistent.  So for persistent volumes the point of creation/deletion of map entry in Stage/Unstage and for ephemeral its publish/unpublish.